### PR TITLE
Add Firefox MR1 promo to home page [fix #10167]

### LINKED ISSUE #10167: Update homepage for MR1
Placeholder for work related to updating the homepage for MR1

### PR DESCRIPTION
**Pull request by craigcook**
## Description
Add's a new Firefox promo to the home page (all home pages: EN, FR, DE, and global). There are four variations, randomly selected on page load. We'll rely on CDN caches to approximate randomness in production.

It's behind the switch `firefox-mr1-launch` and otherwise falls back to the old promo. Post-launch we can remove that switch as well as the older promos.

## Issue / Bugzilla link
#10167 

## Testing
http://localhost:8000
https://www-demo4.allizom.org/home-test/ (test page showing all four promos)

### CODE REVIEW

**Comment by craigcook** [comment]
To do: add/update functional tests

File: bedrock/mozorg/templates/mozorg/home/includes/mr1-promo-ryan.html [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      <svg class="c-fxpromo-noodle c-fxpromo-noodle-circle" xmlns="http://www.w3.org/2000/svg" width="44" height="44" viewBox="0 0 44 44">
+        <circle cx="22" cy="22" r="20.5" fill="none" stroke="#3fe1B0" stroke-width="3">
+          <animate attributeName="stroke" values="#3fe1B0;#fff44f;#3fe1B0;" dur="35s" repeatCount="indefinite" />
+        </circle>
+      </svg>
+
+      {{ high_res_img('img/home/2021/ryan.png', {'alt': '', 'width': '570', 'height': '529', 'class': 'c-fxpromo-photo'}) }}
+
+      <p class="c-fxpromo-caption">
+        <span class="c-fxpromo-caption-name">RyanNicole –</span>
```
Comment (stephaniehobson) [inline bedrock/mozorg/templates/mozorg/home/includes/mr1-promo-ryan.html:53]: No space between first and last name here. Just wanted to check that's intentional.

File: media/css/mozorg/home/home-mr1-promo.scss [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+}
+
+.mzp-c-button-download-container {
+    display: block;
+    margin: 0 auto $layout-md;
+}
+
+.c-fxpromo-altcta {
+    display: none;
+    margin: 0 auto $layout-md;
```
Comment (stephaniehobson) [inline media/css/mozorg/home/home-mr1-promo.scss:32]: The text appears more vertically centered next to the image if there is no bottom margin. OTOH, maybe putting the text a bit higher up is intentional, in which case ignore this.

**Review by stephaniehobson — CHANGES_REQUESTED** [review]
Animations are 🕺 at every screen size. 🔥  

I left a couple notes in the code and I have some spacing questions/suggestions which I will leave in separate comments so they can be hidden if they're not relevant or have been addressed.

**Comment by stephaniehobson** [comment]
On some mobile breakpoints the text is centered and the button is left aligned and it looks  a little odd:

Screen Shot 2021-05-26 at 17 24 29

**Comment by stephaniehobson** [comment]
My eye wants the text to left align with the blocks above and below it. This is probably personal preference

Screen Shot 2021-05-26 at 17 25 08

On the other hand, it would be nice to have more space between the text and the image when the page is full size. _Especially_ in German 😬 

Screen Shot 2021-05-26 at 17 25 58

If other people with more design experience know better, or if it's too late to be changing stuff this is not blocking.

**Comment by stephaniehobson** [comment]
Did I mention the photos and animations look amazing AND adapt really nicely to different screen sizes?

File: bedrock/mozorg/templates/mozorg/home/includes/mr1-promo-ryan.html (hunk 1 above)
Comment (craigcook) [reply bedrock/mozorg/templates/mozorg/home/includes/mr1-promo-ryan.html:53]: That's intentional: https://www.msryannicole.com/home

File: bedrock/mozorg/templates/mozorg/home/includes/mr1-promo-gary.html [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+<section class="c-fxpromo c-fxpromo-gary">
+  <div class="mzp-l-content mzp-t-content-xl">
+
+    <div class="c-fxpromo-content">
+      <h2 class="c-fxpromo-title">{{ ftl('home-promo-title-gary') }}</h2>
+      {{ download_firefox_thanks(dom_id='download-primary', download_location='MR1 Promo: Gary') }}
+      <p class="c-fxpromo-altcta"><a href="{{ url('firefox') }}" class="mzp-c-button mzp-t-product mzp-t-xl" data-cta-type="link" data-cta-text="Learn more" data-cta-position="MR1 Promo: Gary">{{ ftl('ui-learn-more') }}</a></p>
+    </div>
+
+    <div class="c-fxpromo-graphic">
```
[the remark this answers is not in the document]
Comment (craigcook) [reply bedrock/mozorg/templates/mozorg/home/includes/mr1-promo-gary.html:14]: Oh yeah, it causes really bad flickering when the animation is layered behind an image. I'll fix that.

**Comment by alexgibson** [comment]
Sorry for the last minute change request here. The EU team weren't aware this was being added to DE/FR until the strings appeared in Pontoon. They would like to make sure that we have the right messaging and supporting content on the homepage to make it a valuable experience for people landing on the page, and they feel right now the promo/messaging doesn't quite fit right for the target market.

Can you please remove this promo for DE/FR?

**Review by stephaniehobson — APPROVED** [review]
R+ 🔥

**Pull request merged**